<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>open3d_qtClass</class>
 <widget class="QMainWindow" name="open3d_qtClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>975</width>
    <height>616</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>registration_qt</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>30</y>
      <width>391</width>
      <height>421</height>
     </rect>
    </property>
    <property name="title">
     <string>4. fine registration(icp)</string>
    </property>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>icp_method</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_icp_method">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>20</y>
       <width>121</width>
       <height>22</height>
      </rect>
     </property>
     <property name="currentText">
      <string>point2plane</string>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <item>
      <property name="text">
       <string>point2point</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>point2plane</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>G-ICP</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="pushButton_icp">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>131</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>icp refinement</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>icp registration matrix</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="textEdit_matrix_icp">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>190</y>
       <width>391</width>
       <height>100</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>icp_iteration</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_icp_iteration">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>50</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>30</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="textEdit_matrix_final">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>320</y>
       <width>391</width>
       <height>100</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>300</y>
       <width>191</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>coarse+icp registration matrix</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_icp_optimation_times">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>80</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>icp_optimation_times</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>reduce_ratio</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_reduce_ratio">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>110</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0.8</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>391</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>1. choose files</string>
    </property>
    <widget class="QPushButton" name="pushButton_target">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>30</y>
       <width>91</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>target pcd</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_source">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>91</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>source pcd</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_visualize">
     <property name="geometry">
      <rect>
       <x>214</x>
       <y>30</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>visualize pcd</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>300</y>
      <width>391</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>3. coarse registration </string>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>221</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>coarse registration matrix</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_fpfh">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>171</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>coarse registration</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="textEdit_matrix_init">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>80</y>
       <width>391</width>
       <height>100</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_merge_save">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>460</y>
      <width>131</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>merge_save</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>130</y>
      <width>391</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>2. sample and  filter</string>
    </property>
    <widget class="QLineEdit" name="lineEdit_voxel_size">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>100</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>2.0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>voxel size</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_visualize_dowmsampled">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>271</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>visualize pcd voxel_downsampled</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_nb_neighbors">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>40</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>50</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Statistical filter:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>nb_neighbors:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>std_ratio:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_std_ratio">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>70</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>2.0</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_source_filter">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>40</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>source filter</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_target_filter">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>70</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>target filter</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>975</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="registration_qt.qrc"/>
 </resources>
 <connections/>
</ui>
